docs: improve Getting Started (#11202)
renovatebot/renovate: docs/usage/getting-started/20-running.md

# Running Renovate

As a Renovate end user, there are two main categories of use:

- You self-host Renovate, e.g. by running the pre-built Docker image, or
- Someone else is hosting Renovate, and you install/configure it for the repositories you choose

If someone else is hosting Renovate for you, or you are using the WhiteSource Renovate App on GitHub, then you can skip ahead to the Installing Renovate into Repositories section.

## Self-Hosting Renovate

Self-hosting Renovate means that you are the "administrator" of the bot, which entails:

- You provide infrastructure for it to run on,
- You provision its global config,

> - You ensure it's running regularly

Reviewer: ```suggestion
- You ensure it's running regularly
- You ensure Renovate bot itself is updated
```

### Available distributions

> #### npm Package (CLI)

Reviewer: ```suggestion
#### npm package (CLI)
```

Renovate's Open Source CLI is built and distributed as the npm package `renovate`.
You can run this directly in any Node.js environment - even via `npx` - and it will process all the repositories it is configured with, before exiting.

> If you need to support any non-npm lock files like Bundler, Go Modules or Poetry then you'll need to make sure they are pre-installed in the same environment alongside Renovate.

Reviewer: > If you need to support any non-npm lock files like Bundler, Go Modules or Poetry then you'll need to make sure they are pre-installed in the same environment alongside Renovate.

I think this needs expanding on?

I remember somewhere that if you self-host you must ensure all package managers (say `cargo` and `python` and `go`) that are in use on the repository are available for Renovate bot to use?

The `renovate` npm package is compatible with all of Renovate's supported platforms.

> #### Docker Image

Reviewer: ```suggestion
#### Docker image
```

The `renovate` npm package is also distributed in pre-built Node.js images on Docker Hub (`renovate/renovate`).

The `slim` image contains only Node.js so works if either:

- You do not require any additional package managers, or
- You map the Docker socket into the container so that Renovate can dynamically invoke "sidecar" images when necessary

The "full" image (which `latest` defaults to) contains every package manager which Renovate supports already preinstalled.
This approach works best for many, but does have the following downsides:

- It only contains _one_ version of each language/manager - usually the latest
- It's several gigabytes in size

The `renovate/renovate` Docker images are compatible with all of Renovate's supported platforms.

#### GitHub Action

Renovate's npm tool is also provided as a pre-prepared GitHub Action on [`renovatebot/github-action`](https://github.com/renovatebot/github-action).
Details on how to use it can be found in the repository.

> #### GitLab Runner

Reviewer: ```suggestion
#### GitLab runner
```

Is `Runner` a proper noun, does it refer to a specific capability of GitLab? If it's a proper noun, you can skip this suggestion.

Author: ![image](https://user-images.githubusercontent.com/6311784/128868051-8dabd03c-728b-4a13-a543-d67015d45141.png)
Let's keep it capitalized

It is integrated with WhiteSource's vulnerability detection capabilities and additionally supports the capability of "horizontal" scalability - the ability to configure many Renovate "worker" containers which share a common job queue in order to not conflict with each other.

WhiteSource Remediate supports GitHub Enterprise Server, GitLab self-hosted, and Bitbucket Server.

### Hosting Renovate

Once you have decided on a Renovate distribution, you need to decide where and how to run it.

For the GitHub Action and GitLab Runner approaches, they will naturally run on their respective CI infrastructure.
For the npm package approach or Docker images, you will need some form of VM or container infrastructure to run Renovate on.

In all the above cases you will need to make sure that some form of cron-like capability exists to schedule when Renovate runs.
In general we recommend to run Renovate hourly if possible.

WhiteSource Renovate On-Premises and WhiteSource Remediate both run as long-lived containers so do not need any additional cron-like concept as it is built-in.

> ### Global Config

Reviewer: ```suggestion
### Global config
```

Renovate's server-side/admin config is referred to as its "global" config, and can be specified using either a config file, environment variables, or CLI parameters.

Some config is global-only, meaning that either it is only applicable to the bot administrator or it can only be controlled by the administrator and not repository users.
The bot's global config can include both global as well as non-global configuration options, while user/repo config can only include non-global options.
It is recommended to keep as much of the non-global config as possible in repo config files in order to provide maximum transparency to end users.

> ## Installing Renovate Into Repositories

Reviewer: ```suggestion
## Installing Renovate into repositories
```

# Running Renovate

As a Renovate end user, there are two main categories of use:

- You self-host Renovate, e.g. by running the pre-built Docker image, or
- Someone else is hosting Renovate, and you install/configure it for the repositories you choose

If someone else is hosting Renovate for you, or you are using the WhiteSource Renovate App on GitHub, then you can skip ahead to the Installing Renovate into Repositories section.

## Self-Hosting Renovate

Self-hosting Renovate means that you are the "administrator" of the bot, which entails:

- You provide infrastructure for it to run on,
- You provision its global config,
- You ensure it's running regularly

### Available distributions

#### npm Package (CLI)

Renovate's Open Source CLI is built and distributed as the npm package `renovate`.
You can run this directly in any Node.js environment - even via `npx` - and it will process all the repositories it is configured with, before exiting.

> If you need to support any non-npm lock files like Bundler, Go Modules or Poetry then you'll need to make sure they are pre-installed in the same environment alongside Renovate.

Author: ```suggestion
When you install Renovate from npm it naturally does not come bundled with any third party tools or languages such as Ruby, Python, Composer, Bundler, Poetry, etc.
Therefore if you need Renovate to support any non-npm lock files like Bundler then you'll need to make sure all required third party tools are pre-installed in the same environment alongside Renovate before you run it.
```

Reviewer: Better! 😄 I cannot apply review suggestions, so you'll have to do that yourself! 😉
